### ISSUE TITLE
Sign in with Github

### ISSUE THREAD

**Issue by joshghent** [esfiddle/esfiddle#33]
This is a job that will allow lots of new features in the future!

We need a way for users to be able to sign in with their github accounts. We don't need to do anything more with it at this stage, just a basic login, logout function. The rest will be added in future PR's/issues

**Comment by jagadeesh-** [comment]
Hi, I am first timer and would like to take up this task. Please let me know if any specific details i need to know before taking up this task. Thanks.

**Comment by joshghent** [comment]
@jagadeesh- I think we already have @asiyani working on this. Do you have any experience using passportjs?

**Comment by jagadeesh-** [comment]
@joshghent - I don't have working experience with passportjs. But I am very happy to jump in with something that requires this lib.Thanks.

**Comment by asiyani** [comment]
@joshghent I have implemented Git authentication using passportjs.
following 3 Routes added just for testing...
/auth/github -> this will redirect User to Git login page.
/github/onlyAuthoisedUser -> After authenticating you will come to this page. At the moment it will only show you JSON obj of User model which we will be saving in our database.
/github/logout -> User will log out. 

I have updated index.html file to add these 3 buttons to header just for testing. 

Also dotenv is now added so we can set all our env variables locally on .env file. 
Readme file is updated to reflect this. 

Git APP configuration 
Home URL --> http://localhost:3000
Authorization callback URL  --> http://localhost:3000/auth/github/callback
Need to update Client ID &  Client Secret in to .env file (see sample.env)

I would suggest testing authentication by downloading  [myfork](https://github.com/asiyani/esfiddle) before merging to develop branch.
